src/controllers/institution-controller.js

```diff
@@ -68,6 +68,92 @@ export async function addInstitutionToProject(req, res) {
   }
 }
 
+export async function editInstitution(req, res) {
+  const projectId = req.params.projectId
+  const institutionId = req.body.institutionId
+  const name = req.body.name
+  const selectedInstitutionId = req.body.selectedInstitutionId
+
+  const institution = await models.Institution.findByPk(institutionId)
+
+  let presentInstitutionWithName = selectedInstitutionId
+    ? await models.Institution.findByPk(selectedInstitutionId)
+    : await models.Institution.findOne({ where: { name: name } })
+
+  const dupeProjectInstitutionIds =
+    await institutionService.getInstitutionProjectReferences(projectId, [
+      institutionId,
+    ])
+  const institutionxUserIds =
+    await institutionService.getInstitutionUserReferences([institutionId])
+
+  const institutionInUse =
+    dupeProjectInstitutionIds.length || institutionxUserIds.length
+
+  if (presentInstitutionWithName == null && !institutionInUse) {
+    institution.name = name
+    await institution.save({
+      user: req.user,
+    })
+    return res.status(200)
```

Comment:
You have to return the institution in the response so that the client can update its value. Same with the return at the bottom too.